src/deserialize/deserializer.rs

```diff
@@ -211,3 +241,28 @@ impl<'de> Visitor<'de> for JsonValue {
         Ok(nonnull!(dict_ptr))
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct DeserializeError<'a> {
+    pub message: String,
+    pub line: usize,  // start at 1
+    pub column: usize,  // start at 1
+    pub data: &'a str,
+}
+
+impl<'a> DeserializeError<'a> {
+    /// Return position of the error in the deserialized data
+    pub fn pos(&self) -> usize {
```

Comment:
Could you add a test using all PyUnicode formats? ASCII, latin1, two-byte UTF-8, four-byte UTF-8, to test bytes/chars and equivalent with json.

Reply:
I hope https://github.com/ijl/orjson/pull/169/commits/d73f6f656ccb43336e19d812d05512da33a997fb was kind of what you expected